Create ui-c8y-1019-24-3-user-message-in-device-list-page-is-incorrect-when-no-devices.md (#2075)
SoftwareAG/c8y-docs: content/change-logs/device-management/ui-c8y-1019-24-3-user-message-in-device-list-page-is-incorrect-when-no-devices.md

---
date: ""

> title: User message in device list page is incorrect when no devices are available in the tenant [GRAFT][release/cd]

Author: The Claude AI suggested to use the following title:
```suggestion
title: Improved user message on device list page when no devices are available
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component--KIsStyzM
    label: Device Management app
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: DM-1214
version: 1019.24.3
---

> User message in device list page is incorrect when no devices are available in the tenant [GRAFT][release/cd]

Reviewer: ```suggestion
The layout of the empty state message displayed in data grids across all standard applications has been adjusted and unified. Grid that did not differentiate between no data available and no filter matches or search results are now adjusted to display the corresponding empty state message.

For developers the `emptyStateContext` directive was introduced that injects data source statistics variable into template context.
```

> User message in device list page is incorrect when no devices are available in the tenant [GRAFT][release/cd]

Author: The Claude AI suggested to use the following body:
```suggestion
Previously, when no devices were available in a tenant, the device list page displayed an incorrect user message. With this change, the user message on the device list page has been updated to provide a clear and accurate message when no devices are available in the tenant. This improvement enhances the user experience by providing precise feedback on the device availability status.
```

---
date: ""

> title: Improved user message on Device list page when no devices are available

Reviewer 2: ```suggestion
title: Improved layout and content of empty state messages
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component--KIsStyzM
    label: Device Management app
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: DM-1214
version: 1019.24.3
---

> The layout of the empty state message displayed in data grids across all standard applications has been adjusted and unified. Grid that did not differentiate between no data available and no filter matches or search results are now adjusted to display the corresponding empty state message.

Reviewer 3: ```suggestion
The layout of the empty state message displayed in data grids across all standard applications has been adjusted and unified. Grids that did not differentiate between no data available and no filter matches or search results are now adjusted to display the corresponding empty state message.
```

> The layout of the empty state message displayed in data grids across all standard applications has been adjusted and unified. Grids that did not differentiate between no data available and no filter matches or search results are now adjusted to display the corresponding empty state message.

Reviewer 2: @bozhidardedov Is it only the layout that has changed or also the message itself?

> The layout of the empty state message displayed in data grids across all standard applications has been adjusted and unified. Grids that did not differentiate between no data available and no filter matches or search results are now adjusted to display the corresponding empty state message.

Reviewer 2: ```suggestion
The layout of the empty state message displayed in data grids across all standard applications has been adjusted and unified. Moreover, grids that did not differentiate between no data available and no filter matches or search results do now display a more precise corresponding empty state message.
```
